src/OpenSage.Game/Logic/Object/ObjectCreationList.cs

```diff
@@ -100,15 +112,45 @@ internal static CreateDebrisObjectCreationListItem Parse(IniParser parser)
 
         [AddedIn(SageGame.CncGeneralsZeroHour)]
         public int RollRate { get; private set; }
+
+        internal override void Execute(BehaviorUpdateContext context)
+        {
+            var debrisObject = context.GameContext.GameObjects.Add("GenericDebris", context.GameObject.Owner);
+
+            debrisObject.Transform.Translation = context.GameObject.Transform.Translation + Offset;
+            debrisObject.Transform.Rotation = context.GameObject.Transform.Rotation;
+
+            // Model
+            var w3dDebrisDraw = (W3dDebrisDraw) debrisObject.DrawModules[0];
+            // TODO
+            //var modelName = ModelNames[context.GameContext.Random.Next(ModelNames.Length)];
+            var modelName = ModelNames[0];
+            w3dDebrisDraw.SetModelName(modelName);
+
+            // Physics
+            var physicsBehavior = debrisObject.FindBehavior<PhysicsBehavior>();
+            physicsBehavior.Mass = Mass;
+
+            if (Disposition.Get(ObjectDisposition.SendItFlying))
+            {
+                physicsBehavior.AddForce(
+                    new Vector3(
+                        ((float)context.GameContext.Random.NextDouble() - 0.5f) * DispositionIntensity * 200,
```

Comment:
Why does this get multiplied by 200?

Reply:
Because I just used random numbers until I got a decent looking bounce :) I don’t know yet how to calculate these numbers properly.